ipvs.go

```diff
@@ -199,23 +199,37 @@ func parseIPVSBackendStatus(file io.Reader) ([]IPVSBackendStatus, error) {
 	return status, nil
 }
 
-func parseIPPort(s string) (net.IP, uint16, error) {
-	tmp := strings.SplitN(s, ":", 2)
-
-	if len(tmp) != 2 {
-		return nil, 0, fmt.Errorf("invalid IP:Port: %s", s)
-	}
+func parseIPPort(ipPort string) (net.IP, uint16, error) {
+	var (
+		portString string
+		ip         net.IP
+		err        error
+	)
 
-	if len(tmp[0]) != 8 && len(tmp[0]) != 32 {
-		return nil, 0, fmt.Errorf("invalid IP: %s", tmp[0])
+	if len(ipPort) == 13 || len(ipPort) == 37 {
+		tmp := strings.SplitN(ipPort, ":", 2)
```

Comment:
I'm not sure this is true, all of the validation for the `IP:Port` handling is part of this switch block.